docs/app/routes/docs/guides/react-three-fiber.mdx

```diff
@@ -0,0 +1,401 @@
+either, check out our [getting started](/docs/getting-started) or alternatively, the [r3f](https://docs.pmnd.rs/react-three-fiber) docs.
+:::
+
+## Introduction
+
+In this guide we'll explore why `react-spring` is a valuable addition to your `r3f` project, working with the [`imperative API`](/docs/concepts/imperative-api)
+to create performant animation updates on objects in the scene graph and with the [event system](/docs/advanced/events) of the library
+to update parts of your scene that are not wrapped in our [`animated` HOC](/docs/concepts/animated-elements). To work with `react-spring`
+and `react-three-fiber` you'll need to install the `@react-spring/three` package.
+
+```bash
+yarn add @react-spring/three
+```
+
+## Why use the library?
+
+A common question asked is why use `react-spring` with `react-three-fiber` when you can use the `useFrame` hook to update your meshes & objects
+instead without knowing another API. Well this is a great question, lets consider a simple use-case.
+
+I have a distortion blob and I want to it to change color on click. You could do this with `useFrame` to perform frame by frame updates, `useRef`
+to access the material object and use the `THREE.Color.lerp` function, slowly incrementing by lerp value until the color is reached. But this then
+requires that I keep track of a `THREE` instance of `Color` which _can_ lead to memory leaks if not handled correctly and not only that, but it would
+be a lot of code. Then you'd need to think about writing your own easing functions. With `react-spring` you can do this in a few lines of code.
+
+```tsx live=true template=r3f showLineNumbers lines=9,12,20 defaultOpen=true
+import { useState } from 'react'
+import { useSpring, animated } from '@react-spring/three'
+import { MeshDistortMaterial } from '@react-three/drei'
+import { Canvas } from '@react-three/fiber'
+
+const AnimatedMeshDistortMaterial = animated(MeshDistortMaterial)
+
+const MyScene = () => {
+  const [clicked, setClicked] = useState(false)
+
+  const springs = useSpring({
+    color: clicked ? '#569AFF' : '#ff6d6d',
+  })
+
+  const handleClick = () => setClicked(s => !s)
+
+  return (
+    <mesh onClick={handleClick}>
+      <sphereGeometry args={[1.5, 64, 32]} />
+      <AnimatedMeshDistortMaterial
+        speed={5}
+        distort={0.5}
+        color={springs.color}
+      />
+    </mesh>
+  )
+}
+
+export default function MyComponent() {
+  return (
+    <Canvas>
+      <ambientLight intensity={0.8} />
+      <pointLight intensity={1} position={[0, 6, 0]} />
+      <MyScene />
+    </Canvas>
+  )
+}
+```
+
+## Use the imperative API
+
+In the example below, we use the power of the `react-spring`'s api to lean into the imperative requirements
+of working with performant webGL scenes. The blob follows you round the canvas (lines 70-82) & scales on
+interaction (lines 46-56) without a single react render to cause these updates.
```

Comment:
Is it possible to make `lines 70-82` and `lines 46-56` linkable?

Reply:
No unfortunately not, the best i could do is highlight them.